push_hono_images.sh

```diff
@@ -44,8 +46,17 @@ echo "called as $ME"
 if [[ "push_hono_native_images.sh" == "$ME" ]]
 then
   IMAGES=${NATIVE_IMAGES}
+  COMMAND_ROUTER_IMAGE="eclipse/hono-service-command-router-infinispan-native:${TAG}"
+  COMMAND_ROUTER_IMAGE_LEGACY="eclipse/hono-service-command-router-native:${TAG}"
+else
+  COMMAND_ROUTER_IMAGE="eclipse/hono-service-command-router-infinispan:${TAG}"
+  COMMAND_ROUTER_IMAGE_LEGACY="eclipse/hono-service-command-router:${TAG}"
```

Comment:
You should use the `${REPO}` variable to create the image names.
It would also be nice if we could define constants for the involved image names instead of duplicating them all over the script.

Reply:
> It would also be nice if we could define constants for the involved image names instead of duplicating them all over the script.

Not quite sure how you want that done but going to assume you want me to create a `COMMAND_ROUTER_IMAGE_BASE` variable with `hono-service-command-router` as its content and then use that inside the if/else to define the `COMMAND_ROUTER_IMAGE` and `COMMAND_ROUTER_IMAGE_LEGACY` variables.

Let me know if that's sufficient or whether I'm misunderstanding your intentions here.

Reply:
> You should use the `${REPO}` variable to create the image names.

I'm not quite sure how that's supposed to work to be honest.

If the script is run and both CR (container registry) and REPOSITORY are provided as arguments, let's say `CR=foo`, `REPOSITORY=bar` and `VERSION=1.2.3`, then the script would try to do the following in order to create the legacy command-router image:

```
docker tag bar/hono-service-command-router-infinispan:1.2.3 bar/hono-service-command-router:1.2.3
```

But unless I'm missing some information about how the images are built, the `bar/hono-service-command-router-infinispan:1.2.3` image will not exist from the Maven build.

The for loop that iterates through the IMAGES array is hard coding `eclipse/` as the repository to define the `ECLIPSE_IMAGE_NAME` value for each image and then tagging that `eclipse/* and I assume that the idea is that the images that are built on the local machine will always have the `eclipse/` repository from the (maven) build process.

So I would think, in order to be sure that the `hono-service-command-router-infinispan` image that got created by the build process is also tagged as `hono-service-command-router`, I should be using the `eclipse/` repository prefix here

Reply:
Maybe it's clearer if the variable names were named `ECLIPSE_COMMAND_ROUTER_IMAGE_NAME` and `ECLIPSE_COMMAND_ROUTER_LEGACY_IMAGE_NAME`?

I think the misunderstanding might be that with this I'm only doing a **local** re-tagging of the command-router image that was built to make sure that the for loop that then pushes the images to the official registry has both the new image name as well as the legacy image name available.

Reply:
I've pushed some changes to the script to try to make the intents clearer.

Let me know if you still have reservations about the approach or if anything is still unclear.